core/src/main/java/nl/nn/adapterframework/webcontrol/api/TestPipeline.java

```diff
@@ -111,13 +111,19 @@ public Response postTestPipeLine(MultipartBody inputDataMap) throws ApiException
 		}
 
 		if (StringUtils.isNotEmpty(message)) {
+			result.put("message", message);
 			try {
 				PipeLineResult plr = processMessage(adapter, message, secLogMessage);
-				result.put("state", plr.getState());
-				result.put("message", message);
-				result.put("result", plr.getResult().asString());
+				try {
+					result.put("state", plr.getState());
+					result.put("result", plr.getResult().asString());
+				} catch (Exception e) {
+					result.put("state", "Error");
+					result.put("result", "An Exception occurred while extracting the result of the PipeLine with exit state ["+plr.getState()+"]: ("+e.getClass().getTypeName()+") "+e.getMessage());
+				}
 			} catch (Exception e) {
-				throw new ApiException("exception on sending message", e);
+				result.put("state", "Error");
+				result.put("result", "An Exception occurred while processing the message: ("+e.getClass().getTypeName()+") "+e.getMessage());
```

Comment:
Als je wilt dat je dit ook in de gui ziet moet je `Error` vervangen door `ERROR`.
Of je gooit een ApiException, die doet namelijk het zelfde en dan is het uniform door de hele Api.